src/librbd/internal.cc

```diff
@@ -3587,6 +3691,11 @@ namespace librbd {
 	bl.append(buf + q->first, q->second);
       }
 
+      r = ictx->update_object_map(p->objectno, OBJECT_EXISTS);
```

Comment:
these updates in the i/o path should be async still - maybe they'd be better as part of the AioRequest state machines? I thought you had something like that before, was there a reason it wasn't a good idea?